storage/innobase/buf/buf0buf.cc

```diff
@@ -1394,17 +1394,14 @@ inline bool buf_pool_t::withdraw_blocks()
 
 		/* reserve free_list length */
 		if (UT_LIST_GET_LEN(withdraw) < withdraw_target) {
-			buf_flush_LRU(
-				std::max<ulint>(withdraw_target
-						- UT_LIST_GET_LEN(withdraw),
-						srv_LRU_scan_depth),
-				true);
```

Comment:
Because this code was in part replaced by the assignment `try_LRU_scan = false`, `need_LRU_eviction()` will hold and the `buf_flush_page_cleaner()` is guaranteed to make some progress.

Buffer pool resizing can cause the server to hang. Before #3107 there is no logic to abort the buffer pool resizing in case the desired new buffer pool is too small for the workload.